bon-macros/src/builder/builder_gen/input_func.rs

```diff
@@ -503,3 +504,27 @@ impl Visit<'_> for FindSelfReference {
         }
     }
 }
+
+fn get_must_use_attribute(attrs: &[syn::Attribute]) -> Option<syn::Attribute> {
+    let mut iter = attrs
+        .iter()
+        .filter(|attr| !attr.is_doc() && matches!(attr.style, syn::AttrStyle::Outer))
+        .filter(|attr| {
+            let path = match &attr.meta {
+                syn::Meta::Path(path) => path,
+                syn::Meta::List(_) => return false,
+                syn::Meta::NameValue(name_value) => &name_value.path,
+            };
+            path.segments.len() == 1
+                && path
+                    .segments
+                    .iter()
+                    .any(|segment| segment.ident.raw_name() == "must_use")
```

Comment:
Let's follow the pattern of adding extension trait methods to `syn::Attribute` like here: https://github.com/elastio/bon/blob/741ed74217e922e3b6e0fa4480b59f5469469880/bon-macros/src/util/attrs.rs#L11-L21

Btw. a simpler way to check that `syn::Path` is equal to `must_use` is with the [`syn::Path::is_ident()`](https://docs.rs/syn/latest/syn/struct.Path.html#method.is_ident) method.